relay-server/src/actors/project_local.rs

```diff
@@ -115,9 +144,19 @@ fn load_local_states(projects_path: &Path) -> io::Result<HashMap<ProjectKey, Arc
             continue;
         }
 
-        let state = serde_json::from_reader(io::BufReader::new(fs::File::open(&path)?))?;
-        let mut sanitized = ProjectState::sanitize(state);
+        fn parse_file(
+            path: std::path::PathBuf,
+        ) -> tokio::io::Result<(std::path::PathBuf, ProjectState)> {
+            let file = std::fs::File::open(&path)?;
+            let reader = std::io::BufReader::new(file);
+            Ok((path, serde_json::from_reader(reader)?))
+        }
 
+        // serde_json is not async, so spawn a blocking task here:
+        let handle = tokio::task::spawn_blocking(move || parse_file(path));
+        let (path, state) = handle.await??;
```

Comment:
I would simply inline this 
```suggestion
        let (path, state) = tokio::task::spawn_blocking(move || parse_file(path)).await??;
       
```